### ISSUE TITLE
Return cached object from a subroutine [@xloil.func(command=True)]

### ISSUE THREAD

**Issue by talalvas** [cunnane/xloil#49]
Hello,
I'm using the powerful option to return cached objects from UDFs.

My problem is when I try to return cached objects from subroutines, it seems that it doesn't work properly.

```python
@xlo.func(command=True)
def xlo_xrt_df():

    wb = xlo.active_workbook()
    xrt_df_rng = xlo.Range(wb.Names['priExchangeRates'].RefersTo) # Defines the range to return the Dataframe

    try:
        qry_df = get_xrt_dataframe() # Returns a pandas Dataframe
    except Exception as e:
        xrt_df_rng.value = str(e)
    
    if qry_df.empty:
        xrt_df_rng.value = xlo.CellError.NULL
    else:
        xrt_df_rng.value = xlo.cache(qry_df)
```

The function returns a value to the cell that isn't representing a sheet, column and a row like cached values returned from UDFs.
The returned value is a string and not a cached dataframe.

instead of:
`欣[xrt_wb.xlsm]sheet1!R3C4,A`
The function returns:
`欣xx,A`

My question is: Is it even possible to return a cached object from such a function? Or this capability is limited only for UDFs?

Thanks.

**Comment by cunnane** [comment]
It's certainly possible to create a cache object anywhere, however due to a bug the cache string was somewhat truncated as you observed, although you should be able to retrieve the object using that string.  There are some caveats to using the object cache outside of UDFs.  I have fixed the bug in v0.17.10 have released that version.  I've also [expanded the docs](https://xloil.readthedocs.io/en/stable/xlOil_Python/TypeConversion.html#cached-objects) to explain exactly what those caveats are.   Let me know if the explanation makes sense to you.

**Comment by talalvas** [comment]
Thank you for you replay.
After updating to the latest version, the problem has been solved.
The subroutine returns a cached object with the correct cache string, and the cached object acts properly.

Keep up your amazing work! It is very much appreciated.
